test/parallel/test-child-process-spawn-controller.js

```diff
@@ -39,3 +39,49 @@ const cp = require('child_process');
     assert.strictEqual(e.name, 'AbortError');
   }));
 }
+
+{
+  // Verify that passing kill signal works.
+  const controller = new AbortController();
+  const { signal } = controller;
+
+  const echo = cp.spawn('echo', ['fun'], {
+    encoding: 'utf8',
+    shell: true,
+    signal,
+    killSignal: 'SIGKILL',
+  });
+
+  echo.on('close', common.mustCall((code, killSignal) => {
+    assert.strictEqual(killSignal, 'SIGKILL');
+  }));
+
+  echo.on('error', common.mustCall((e) => {
+    assert.strictEqual(e.name, 'AbortError');
+  }));
+
+  controller.abort();
+}
+
+{
+  // Verify that passing a different kill signal works.
+  const controller = new AbortController();
+  const { signal } = controller;
+
+  const echo = cp.spawn('echo', ['fun'], {
+    encoding: 'utf8',
+    shell: true,
+    signal,
+    killSignal: 'SIGTERM',
+  });
+
+  echo.on('close', common.mustCall((code, killSignal) => {
+    assert.strictEqual(killSignal, 'SIGTERM');
```

Comment:
hmm, is this the bug?

Reply:
Yes, this is what I talked to you about separately. I have a feeling that the existing tests are not actually testing that the abort signal is working correctly. I think that the process is closing by itself (it's a short-lived echo) but the signal is still saying that it closed the process (when it didn't) and that's why the killSignal is actually null and not 'SIGTERM'.

Reply:
I'll try to open a separate issue later today/see if I can fix this with a PR

Reply:
Ok, then is this PR blocked on that one (since the test is failing)?

Reply:
Technically the two features are separate (as this PR is unrelated to abort signals, just by chance because they also use killSignal internally) but I think that it makes more sense to fix the issue before merging this.

Reply:
Ok, marking this as "blocked" on that in the meantime

Reply:
@benjamingr blocked can now be removed